services/QuillLMS/lib/tasks/diagnostic_question_optimal_concepts.rake

```diff
@@ -0,0 +1,156 @@
+# frozen_string_literal: true
+
+namespace :diagnostic_question_optimal_concepts do
+  DELIMITER = "\n"
+
+  DIAGNOSTIC_INDEX = 0
+  QUESTION_INDEX = 2
+  QUESTION_NUMBER_INDEX = 6
+  OPTIMAL_CONCEPTS_INDEX = 3
+
+  DIAGNOSTIC_ID_LOOKUP = {
+    "Starter Pre" => 1663,
+    "Starter Post" => 1664,
+    "Intermediate Pre" => 1668,
+    "Intermediate Post" => 1669,
+    "Advanced Pre" => 1678,
+    "Advanced Post" => 1680,
+    "ELL Starter Pre" => 1161,
+    "ELL Starter Post" => 1774,
+    "ELL Intermediate Pre" => 1568,
+    "ELL Intermediate Post" => 1814,
+    "ELL Advanced Pre" => 1590,
+    "ELL Advanced Post" => 1818,
+    "Springboard" => 1432,
+    "PreAP 1" => 1229,
+    "PreAP 2" => 1230,
+    "AP" => 992
+  }
+
+  desc 'Populate diagnostic_question_optimal_concepts from CSV extracted from Curriculum Notion database: https://www.notion.so/quill/f937b1cc8d2d4ed8943a36ce35dca177?v=7dd55ee1f04e4b5daf79645aae960aa1'
```

Comment:
I think it's worth doing a refactor pass on these rake tasks. They are very long and complicated, and a bit hard to review. The validation script doesn't share any code with the live script. You can use modules to define sub methods for use in Rake tasks, which might help clean this up a bit.

Reply:
You should also consider having a `dry_run` or `live` flag` on the main script that you could use for validation purposes (so that your validation uses the same code as your live script).